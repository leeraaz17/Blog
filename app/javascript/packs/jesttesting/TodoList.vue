<template>
  <div class="todo-list">
    <div v-if="listEmpty">You have nothing else to do :(</div>
    <ul v-else>
      <li v-for="item in todos" :key="item.id">{{ item.text }}</li>
    </ul>
  </div>
</template>

<script>
  export default {
    props: {
      todos: {
        type: Array,
        default: () => [],
        validator: value =>
          Array.isArray(value) && value.every(item => item.id && item.text)
      }
    },
    computed: {
      listEmpty() {
        return this.todos.length === 0
      }
    }
  }
</script>