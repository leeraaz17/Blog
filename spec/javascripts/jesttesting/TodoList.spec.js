import { shallowMount, mount } from '@vue/test-utils';
import TodoList from '@/app/javascript/packs/jesttesting/TodoList.vue';

describe('TodoList', () => {
  it('should render to a snapshot when list is empty', () => {
    const propsData = { todos: [] }
    const wrapper = shallowMount(TodoList, { propsData })
    // const jpt = {beef: "hehe", bacon: "haha"}
    // console.log(JSON.parse(JSON.stringify(jpt)))
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render to a snapshot when list has items', () => {
    const propsData = { todos: [{ id: 1, text: 'Todo #1' }] }
    const wrapper = shallowMount(TodoList, { propsData })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('is a Vue instance', () => {
    const wrapper = mount(TodoList)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
})