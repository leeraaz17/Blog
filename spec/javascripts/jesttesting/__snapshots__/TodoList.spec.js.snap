// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should render to a snapshot when list has items 1`] = `
<div
  class="todo-list"
>
  <ul>
    <li>
      Todo #1
    </li>
  </ul>
</div>
`;

exports[`TodoList should render to a snapshot when list is empty 1`] = `
<div
  class="todo-list"
>
  <div>
    You have nothing else to do :(
  </div>
</div>
`;
